recipe/build.sh

```diff
@@ -158,3 +158,9 @@ if [[ ("$target_platform" != "linux-ppc64le") && \
   # https://forum.hdfgroup.org/t/hdf5-1-10-long-double-conversions-tests-failed-in-ppc64le/4077
   make check RUNPARALLEL="mpiexec -n 2"
 fi
+
+# test for hdf5 C++ exceptions
+if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" != "1" || "${CROSSCOMPILING_EMULATOR}" != "" ]]; then
+    $CXX ${CXXFLAGS} $RECIPE_DIR/testhdf5exc.cpp -o testhdf5exc -L$PREFIX/lib -lhdf5_cpp -lhdf5 -I$PREFIX/include
```

Comment:
```suggestion
    ${CXX} ${CXXFLAGS} ${LDFLAGS} $RECIPE_DIR/testhdf5exc.cpp -o testhdf5exc -lhdf5_cpp -lhdf5 -I$PREFIX/include
```

You can also follow the style and add the test to the test section:
https://github.com/xylar/hdf5-feedstock/blob/add-exception-catch-test/recipe/run_test.sh#L23

Reply:
Agreed, I will move it if we decide we want to keep the test.